### ISSUE TITLE
aws: limit the scope of IAM instance policy

### ISSUE THREAD

**Issue by vaijab** [UKHomeOffice/keto#100]
Current AWS implementation allows for compute and master nodes to manage pretty much all EC2 resources across AWS account, see:
https://github.com/UKHomeOffice/keto/blob/master/pkg/cloudprovider/providers/aws/cloudformation_templates.go#L356
https://github.com/UKHomeOffice/keto/blob/master/pkg/cloudprovider/providers/aws/cloudformation_templates.go#L533

### compute nodes
- should only be able to describe their own tags. I cannot think of anything else that they need?

### master nodes
- whatever kubernetes requires
- should be able to set tags on compute nodes as part of the same cluster

**Comment by jon-shanks** [comment]
Makes sense to me , we definitely should limit the compute nodes. Would be good to limit the master but less of a security risk to a degree but probably worth investing the effort in both
